mysensors/mysensors.py

```diff
@@ -49,6 +50,22 @@ def __init__(self, event_callback=None, persistence=False,
         if persistence:
             self._safe_load_sensors()
 
+    @property
+    def event_callback(self):
+        """Getter for event_callback property."""
+        return self._event_callback
+
+    @event_callback.setter
+    def event_callback(self, callback):
+        """Setter for event_callback property."""
+        self._event_callback = callback
+        if callback is not None:
+            if len(inspect.signature(callback).parameters) == 2:
+                def adapter(message):
+                    """Event callback adapter for backwards compatibility."""
+                    callback('sensor_update', message.node_id)
+                self._event_callback = adapter
```

Comment:
Even with this adapter the call signature for the `event_callback` attribute of the gateway instance will have changed, so it's not possible to call that the same way as before. I don't think we should assume that we're only going to call the `event_callback` attribute from inside the class. I think the adapter makes things more confusing for a user.

If we think the added info from passing the full message instance as argument to the callback is worth a breaking change, I suggest we just do that, and update the readme clearly and announce a breaking change.